src/PhoneNumberInput/PhoneNumberInput.jsx

```diff
@@ -116,15 +132,32 @@ const PhoneNumberInput = forwardRef((props, ref) => {
     onCountryNumberChange(country)
   }
 
+  const onCursorPositionChanged = () => {
+    setCursorPositionChangedToggler(!cursorPositionChangedToggler)
+  }
+
   const onCountryNumberChange = country => {
-    const value = `${country.prefix}${displayValue}`
+    const value = `${country.prefix}${trimStartingZero(displayValue)}`
     const event = { target: { value } }
     onChange(event)
   }
+
   const onNumberChange = event => {
-    event.target.value = `${country.prefix}${event.target.value}`
+    if (trimStartingZero(event.target.value) !== event.target.value) {
+      setCursorPosition(0)
+    } else {
+      setCursorPosition(event.target.selectionStart)
+    }
+
+    onCursorPositionChanged()
```

Comment:
What if I'm pressing Cmd+C or Shift+A? And what about while selecting a portion of the number? You are modifying the functionality of the input element in ways you cannot always predict.

Reply:
Ctrl+A and Ctrl+C are not affected.